transport/tchannel.go

```diff
@@ -118,15 +152,28 @@ func (t *tchan) readResponse(call *tchannel.OutboundCall) (*Response, error) {
 	}, nil
 }
 
-func (t *tchan) writeArgs(call *tchannel.OutboundCall, headers map[string]string, body []byte) error {
+func (t *tchan) writeArgs(call *tchannel.OutboundCall, r *Request) error {
 	if err := writeHelper(call.Arg2Writer, func(writer tchannel.ArgWriter) error {
-		return thrift.WriteHeaders(writer, headers)
+		switch t.callOptions.Format {
+		case tchannel.JSON:
+			encoder := json.NewEncoder(writer)
+			return encoder.Encode(r.Headers)
+		case tchannel.Raw:
+			if v, ok := r.Headers[rawHeadersKey]; ok {
+				_, err := io.WriteString(writer, v)
```

Comment:
Requests is immutable, and if the request was used with a benchmark, the same raw header has to be sent multiple times, so we have to keep it